### PR TITLE
Bump anchor

### PR DESCRIPTION
**Pull request by ChewingGlass**
Bumping to the latest version of anchor

### CODE REVIEW

File: libraries/concurrent-merkle-tree/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ log = []
 sol-log = [ "log" ]
 
 [dependencies]
-solana-program = "1.10.33"
+solana-program = "1.16"
```
Comment (2501babe) [inline libraries/concurrent-merkle-tree/Cargo.toml:15]:
```suggestion
solana-program = "1.16.3"
```

**Comment by 2501babe** [comment]
cc @ngundotra for review in case theres a reason to be on 1.10. the 1.14 -> 1.16 change in anchor and the rest of spl also entailed a borsh version change which breaks builds. so getting everything on one version will hopefully help downstream

**Comment by ngundotra** [comment]
@2501babe I want to approve this, but with the version bump the `account-compression/sdk` tests no longer pass. They throw the following error: `Invoked an instruction with too many accounts (14693582971755349667 > 255)`

**Comment by 2501babe** [comment]
lol im having problems just building it myself because anchor 0.28 is pulling in borsh 0.9 instead of 0.10 because it has a too-permissive version constraint, probably as a hack to be able to keep people on 1.14...

**Comment by ngundotra** [comment]
@ChewingGlass is this even building for you?

**Comment by 2501babe** [comment]
the Cargo.lock file is bad for me, it should build if you delete it and let it be regenerated

oh my god i wrote out a whole thing about the issues ive been having getting the tests to connect to the validator properly but its because `localhost` was routing over ipv6 lol... im getting the too many accounts error as expected now and should have time to try and debug soon-ish

**Comment by ChewingGlass** [comment]
It was building for me, yes. But I ended up giving up on the path of forking and upgrading everyone. The Borsh upgrade just breaks too much stuff. Anchor-gen ended up being the solution. Feel free to close this PR and not do the upgrade.

**Comment by ngundotra** [comment]
Closed by #4857

**Pull request closed without merging**